drivers/slipdev/slipdev.c

```diff
@@ -73,6 +73,23 @@ static void _slip_rx_cb(void *arg, uint8_t byte)
     }
 }
 
+static void _poweron(slipdev_t *dev)
+{
+    if ((dev->state != SLIPDEV_STATE_STANDBY) ||
+        (dev->state != SLIPDEV_STATE_SLEEP)) {
+        return;
+    }
+
+    dev->state = 0;
+    uart_init(dev->config.uart, dev->config.baudrate, _slip_rx_cb, dev);
+}
+
+static void _poweroff(slipdev_t *dev, uint8_t state)
+{
+    uart_poweroff(dev->config.uart);
+    dev->state = state;
+}
```

Comment:
Couldn't this potentially lead to a synchronization issue with the peer? As it stands, it basically assumes that it will start a frame anew. However, what if the peer is currently in a frame and waiting for the END byte?

Reply:
Do you mean this could lead to problems if we put the UART to sleep while a frame is being received? 
I guess we could discard the current frame in `_poweron()`

Reply:
> Do you mean this could lead to problems if we put the UART to sleep while a frame is being received?

No, I meant while a frame is being sent. Or is this prevented somehow?

> I guess we could discard the current frame in `_poweron()`

Discarding an incompletely received frame might be a good idea regardless ;-). But I think this is implicitly already happening by setting `state` to 0 in `_poweron()`.

Reply:
> Both `.send()` and `.set()` functions are executed by the netdev thread, so they can not happen in parallel.

IIRC that might not be the case, once the MAC rework is done...

Reply:
this would then also break that assumption in all other drivers…

Reply:
Ok. Then I guess this is fine :-).